### PR TITLE
[Core] Add syntax check for --set parameter in generic update command

### LINKED ISSUE #10464: Failure in running az vmss update on diskprofile
## Describe the bug

**Command Name**
`az vmss update`

**Errors:**
```
pop from empty list
Traceback (most recent call last):
Temp\pip-install-qxmmnr17\knack\knack\cli.py, ln 206, in invoke
azure\cli\core\commands\__init__.py, ln 603, in execute
azure\cli\core\commands\__init__.py, ln 661, in _run_jobs_serially
azure\cli\core\commands\__init__.py, ln 654, in _run_job
Local\Temp\pip-install-qxmmnr17\six\six.py, ln 693, in reraise
azure\cli\core\commands\__init__.py, ln 631, in _run_job
azure\cli\core\commands\__init__.py, ln 305, in __call__
azure\cli\core\commands\arm.py, ln 525, in handler
azure\cli\core\commands\arm.py, ln 806, in set_properties
azure\cli\core\commands\arm.py, ln 968, in _get_name_path
IndexError: pop from empty list
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az vmss update -g {} -n {} --set {} {} {}`

## Environment Summary
```
Windows-10-10.0.18362-SP0
Python 3.6.6
Shell: cmd.exe

azure-cli 2.0.72

```

### PR DESCRIPTION
**Pull request by qwordy**
[Core] Add syntax check for `--set` parameter in generic update command

Fix #10464

A user misuses this parameter.

Original error message:
```
IndexError: pop from empty list
```

Error message after this patch:
```
usage error: Empty key in --set. Correct syntax: --set KEY=VALUE [KEY=VALUE ...]
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 1]
```diff
@@ -796,6 +796,9 @@ def _find_split():
 def set_properties(instance, expression, force_string):
     key, value = _split_key_value_pair(expression)
 
+    if key.strip() == '':
```
Comment (yonzhan) [inline src/azure-cli-core/azure/cli/core/commands/arm.py]: Can key be None object ?

Comment (Juliehzl) [reply src/azure-cli-core/azure/cli/core/commands/arm.py]: > Can key be None object ?

If key is None type, key.strip() will raise AttributeError.

Comment (Juliehzl) [reply src/azure-cli-core/azure/cli/core/commands/arm.py]: I have one question about the key value. Although I cannot see the debug information, from your conversation in issue #10464, the error is cause by the space in key value, right? If yes, can we do some check in command/argument validation?

Comment (qwordy) [reply src/azure-cli-core/azure/cli/core/commands/arm.py]: > 
> 
> Can key be None object ?

Yes. We should pay attention to corner case.

Comment (qwordy) [reply src/azure-cli-core/azure/cli/core/commands/arm.py]: > 
> 
> I have one question about the key value. Although I cannot see the debug information, from your conversation in issue #10464, the error is cause by the space in key value, right? If yes, can we do some check in command/argument validation?

Yes, we can. But it's a generic update command and set parameter. The logic has been already defined in arm.py in core module.

**Pull request merged**